### PR TITLE
Feat: delete failing profiles from permissions

### PR DESCRIPTION
**Pull request by VirginiaBalseiro**
# Delete failing profiles from permissions

- Add "try again" button to error message
- Add "remove" button next to webId
- Add tests

# To test:
- Add a 'wrong' webId (for instance, with a typo) to a resource's permissions

# Checklist

- [x] All acceptance criteria are met.
- [ ] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: components/resourceDetails/resourceSharing/agentAccessList/index.jsx [hunk 1]
```diff
@@ -32,7 +34,7 @@ const useStyles = makeStyles((theme) => createStyles(styles(theme)));
 export const TESTCAFE_ID_AGENT_ACCESS_LIST_SHOW_ALL =
   "agent-access-list-show-all";
 
-function AgentAccessList({ onLoading }) {
+function AgentAccessList({ onLoading, ...buttonProps }) {
```
Comment (megoth) [inline components/resourceDetails/resourceSharing/agentAccessList/index.jsx:37]: It's a bit weird that the rest of the properties for `AgentAccessList` is called `buttonProps` - the name of the component indicates that it's a list, after all. Further, the buttonProps is only used for a specific part of the component, which might cause confusion if we need to extend other elements.

I suggest adding buttonProps as proper property, and rename it to be more specific to how it's applied, e.g. `({ accessButtonProps, onLoading })`.

**Review by megoth — CHANGES_REQUESTED** [review]
I would've liked the tests for AgentAccess and AgentAccessList components turn green, but I'll only ask that you refactor buttonProps in AgentAccessList.

**Review by megoth — APPROVED** [review]
Great work! The only thing I would change is call it `buttonClassName` instead of `buttonClasses`.

**Comment by VirginiaBalseiro** [comment]
> Great work! The only thing I would change is call it `buttonClassName` instead of `buttonClasses`.

done :)

**Review by megoth — APPROVED** [review]

**Review by ajacksified — APPROVED** [review]

File: components/resourceDetails/resourceSharing/agentAccess/index.jsx [hunk 2]
```diff
@@ -93,26 +109,118 @@ export function getDialogId(datasetIri) {
...
+    } catch (error) {
+      profileError = error;
+      setIsLoading(false);
+    }
+  };
+
   if (profileError) {
-    const message = `Failed to load ${webId}`;
-    return <Message variant="error">{message}</Message>;
+    const message = FAILED_PROFILE_MESSAGE;
```
Comment (ajacksified) [inline components/resourceDetails/resourceSharing/agentAccess/index.jsx:163]: Why create this variable instead of using `FAILED_PROFILE_MESSAGE` directly?

**Review by ajacksified — APPROVED** [review]
Just one minor comment - looks good!

File: components/resourceDetails/resourceSharing/agentAccess/index.jsx (hunk 2 above)
Comment (VirginiaBalseiro) [reply components/resourceDetails/resourceSharing/agentAccess/index.jsx:163]: It seems I was planning to use it for test purposes but didn't. Removed.

**Review by kevin-inrupt — APPROVED** [review]

**Pull request merged**